alarm_trap_sender.cpp

```diff
@@ -182,11 +196,52 @@ void AlarmTrapSender::sync_alarms(bool do_clear)
   }
 }
 
+static int alarm_trap_send_callback(int op,
+                                    snmp_session* session,
+                                    int req_id,
+                                    snmp_pdu* pdu,
+                                    void* correlator)
+{
+  AlarmTableDef* alarm_table_def = (AlarmTableDef *)correlator; correlator = NULL;
+  AlarmTrapSender::get_instance().alarm_trap_send_callback(op,
+                                                           *alarm_table_def);
+  delete alarm_table_def; alarm_table_def = NULL;
+  return 1;
+}
+
+void AlarmTrapSender::alarm_trap_send_callback(int op,
+                                               const AlarmTableDef& alarm_table_def)
+{
+  switch (op)
+  {
+  case NETSNMP_CALLBACK_OP_RECEIVED_MESSAGE:
+    TRC_DEBUG("Alarm successfully delivered");
+    break;
+  case NETSNMP_CALLBACK_OP_SEND_FAILED:
+    // This can happen pretty fast, so don't spin
+    sleep(1);
+    // Fall through to...
+  case NETSNMP_CALLBACK_OP_TIMED_OUT:
+    TRC_DEBUG("Failed to deliver alarm");
+    if (_observed_alarms.is_active(alarm_table_def))
+    {
+      // Alarm is still active, attempt to re-transmit
+      send_trap(alarm_table_def);
```

Comment:
This retransmits to all configured alarm destinations, even if only one is down, right?